src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultClientNotifierService.cs

```diff
@@ -31,7 +31,8 @@ public DefaultClientNotifierService(StreamJsonRpc.JsonRpc jsonRpc)
     public override async Task<TResponse> SendRequestAsync<TParams, TResponse>(string method, TParams @params, CancellationToken cancellationToken)
     {
         await _initializedCompletionSource.Task.ConfigureAwait(false);
-        var result = await _jsonRpc.InvokeAsync<TResponse>(method, @params).ConfigureAwait(false);
+        var arguments = new object[] { @params! };
+        var result = await _jsonRpc.InvokeWithCancellationAsync<TResponse>(method, arguments, cancellationToken).ConfigureAwait(false);
```

Comment:
This was a _pain_ to track down, and reproing it is equally painful, but what I think is happening is that during integartion test startup, we're getting a diagnostics request that is subsequently cancelled by the editor. There is logic in CLaSP, however, that is allowing the task cancellation to bring the queue down because it is not the cancellation token it is expecting, so it assumes something terrible has happened.

This manifested as a `razor/fault` report once per integration test run, I suspect due to the nature of the integration test initialization, where we open a Razor file to verify everything is up and running, and then close it very quickly. In real world usage I never ran into any failures.